Sources/AsyncSequenceValidation/Test.swift

```diff
@@ -116,19 +116,19 @@ extension AsyncSequenceValidationDiagram {
     }()
     
     static var unownedExecutor: UnownedSerialExecutor {
-      (_executor as! any SerialExecutor).asUnownedSerialExecutor()
+      _executor.asUnownedSerialExecutor()
     }
 #endif
 
-    static var clock: Clock?
-    
-    
-    
-    static var driver: TaskDriver?
+    static let clock = LockIsolated<Clock?>(nil)
```

Comment:
Having multiple separate locks here to protect the state is problematic since there might be invariants that aren't enforced this way. I would rather put all the state behind one lock

Reply:
I have modified it to protect all states with a single lock.